lib/rate_map.c

```diff
@@ -235,66 +235,124 @@ rate_map_shift_by_mass(rate_map_t *self, double pos, double mass)
 #error "Base 2 floating point types required"
 #endif
 
+static bool
+valid_sorted_nonempty_array(const double *array, size_t size)
+{
+    size_t idx;
+
+    if ((ptrdiff_t) size < 1 || isnan(array[0])) {
```

Comment:
Clever, I haven't seen this way of detecting a negative size before. But what if we have an array of 2**63 elements?!! :wink:

Reply:
that will only happen with future computers with 128 bit size_t. :)